components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -66,6 +69,12 @@
     //This is used to pass the verifyEmail or askPassword claim from preAddUser to postAddUser
     private static ThreadLocal<Claim> emailVerifyTemporaryClaim = new ThreadLocal<>();
 
+    //Error messages that are caused by password pattern violations
+    private static String[] pwdPatternViolations = new String[]{UserCoreErrorConstants.ErrorMessages
```

Comment:
How can we say these errors can occur only when password policy violations?

Reply:
Change th logic so that it checks for some other substring to verify that this actually is a password pattern violation